### PR TITLE
Kubernetes Response Engine, Part 7: Falcosidekick + Cloud Functions

### LINKED ISSUE #443: Creating Kubernetes Response Engine using GCE and Google Cloud Functions
/area blog

**What would you like to be added**:

Recently falcosidekick got a new output type called [Google Cloud Functions](https://github.com/falcosecurity/falcosidekick/pull/241). In this blog, we will try to demonstrate that how we can use this to implement a new type of Kubernetes Response Engine.

gcloudfalcov2 (1)

### PR DESCRIPTION
**Pull request by Dentrax**
Signed-off-by: Furkan <furkan.turkal@trendyol.com>
Co-authored-by: Batuhan <batuhan.apaydin@trendyol.com>

kre_gke_v3

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind user-interface

/kind content

> /kind translation

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area blog

> /area documentation

> /area videos

**What this PR does / why we need it**:

Please see the linked issue.

**Which issue(s) this PR fixes**:

Fixes #443

**Special notes for your reviewer**:
Currently W.I.P

### CODE REVIEW

**Comment by Dentrax** [comment]
Thank you so much for your support @developer-guy 💫

File: content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md
Comment (radhikapc) [inline content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md:20]: Earlier in this series, we have seen how to use Argo, Tekton, and Knative to trigger a resource after getting input from Falcosidekick.

File: content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md
Comment (radhikapc) [inline content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md:21]: Recently, Falcosidekick received a new output type support for [Cloud Functions](https://github.com/falcosecurity/falcosidekick/pull/241).

File: content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md
Comment (radhikapc) [inline content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md:23]: In this part, let us learn how we can use Falcosidekick and Cloud Functions to detect and delete a compromised pod.

File: content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md
Comment (radhikapc) [inline content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md:25]: We will not go through what Cloud Functions is in-depth, however, you can always find a good overview about it in the [official documentation](https://cloud.google.com/functions).

File: content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md
Comment (radhikapc) [inline content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md:31]: The link is not working.

File: content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md
Comment (radhikapc) [inline content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md:67]: In the beginning, we already enabled [WorkloadIdentity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) feature for our GKE Cluster by setting `--workload-pool` flag. What we need to do here is to add a `iam.workloadIdentityUser` role for the given Service Account.

File: content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md
Comment (radhikapc) [inline content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md:51]: Configure IAM Service Accounts

File: content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md
Comment (radhikapc) [inline content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md:86]: To limit function role access in the particular cluster, ensure that only SA has limited permissions within a particular namespace by using [Role Bindings](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding).

File: content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md
Comment (radhikapc) [inline content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md:130]: It is unclear what you mean by "Where Secret Manager get involved our architecture". Please rewrite for clarity.

**Review by radhikapc — CHANGES_REQUESTED** [review]
Minor changes requested for clarity.

**Comment by Dentrax** [comment]
Thanks for the reviews @radhikapc 🧡 I pushed to the latest version.

**Review by radhikapc — APPROVED** [review]
LGTM

**Comment by developer-guy** [comment]
Hello folks @radhikapc @Dentrax, we have to wait for a release falcosidekick for its helm chart before publishing this blog post on the Falco website, because there is not support for GCP Cloud Functions in the current Falcosidekick Helm Chart yet 🙋🏻‍♂️

**Comment by Dentrax** [comment]
I think this PR is blocking from the following:

- In the beginning of the article, I [referenced](https://github.com/falcosecurity/falco-website/pull/453/files#diff-8a071d693ed114f0770acb26beb3389dcf5d0d41dd2f744528a9391e2c424a27R17) the 6th part (see the pull #446) of the KRE series.
- See, the https://github.com/falcosecurity/charts/pull/239 for the Cloud Function Helm Chart.

File: content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md
Comment (Issif) [inline content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md:16]: Please add part 6 in that list, and part 7 in list of all other posts

Comment (Dentrax) [reply content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md:16]: Thanks, done!

**Review by leogr — COMMENTED** [review]
The Netifly builder complained about the missing `/blog/falcosidekick-reponse-engine-part-6-cloud-run` reference:

```
10:41:36 AM: Building sites … ERROR 2021/06/29 08:41:36 [en] REF_NOT_FOUND: Ref "/blog/falcosidekick-reponse-engine-part-6-cloud-run": "/opt/build/repo/content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md:17:70": page not found
```

It's happening because the part 6 file is missing here. Rebasing this PR on top of the master branch should solve the issue.

**Comment by Dentrax** [comment]
Sure, rebased!

**Review by Issif — APPROVED** [review]

**Comment by developer-guy** [comment]
@radhikapc i think you should approve this PR too, can please approve the PR? 🙋🏻‍♂️🙏

**Review by radhikapc — APPROVED** [review]
Lgtm

**Pull request merged**
